bin/build_update_subfolders.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+import send_email
+import synapseclient
+import synapseutils
+import sys
+
+from typing import Any, List, Tuple, Union
+
+
+def is_new_submitter(syn, submitter_id, parent_folder_id):
+    """
+    Used to check if the submitter is new or already has an existing folder.
+
+    This function uses ``synapseutils.walk`` to crawl through the level 1 subfolders
+    (i.e. The folders 1 level below the Parent-Folder, that are named after the submitter's synapse ID) to
+    verify whether or not a submitter already has an existing folder created.
+    """
+
+    submitter_subfolders = []
+    # Setting ``include_types`` to only Folders so that Files are excluded, and walk is faster.
+    for _, level1_subfolders, _ in synapseutils.walk(syn=syn, synId=parent_folder_id,
+                                                     includeTypes=["folder"]
+                                                     ):
+        for folder_name, synid in level1_subfolders:
+            submitter_subfolders.append(folder_name)
+
+    if submitter_id in submitter_subfolders:
+        return False
+
+    return True
+
+
+def build_subfolder(syn: synapseclient.Synapse, folder_name: str, parent_folder: Union[str, synapseclient.Entity]) -> synapseclient.Entity:
+    """
+    Builds a subfolder under the designated ``parent_folder``.
+
+    Arguments:
+        syn: A Synapse Python client instance
+        folder_name: The name of the subfolder to be created
+        parent_folder: A synapse Id or Entity of the parent folder under which the subfolder will live
+
+    Returns:
+        The created Synapse Folder entity
+    """
+
+    # Create Folder object
+    subfolder = synapseclient.Folder(
+        name=folder_name, parent=parent_folder
+    )
+    subfolder = syn.store(obj=subfolder)
+
+    return subfolder
+
+
+# TODO: https://sagebionetworks.jira.com/browse/IBCDPE-809
+# def update_subfolders():
+#     """"""
+
+
+def update_permissions(syn: synapseclient.Synapse, subfolder: Union[str, synapseclient.Entity],
```

Comment:
Note to self: I think I need to revoke the participating teams' permissions _in addition_ to granting the organizers permissions.